test/algorithms/ground_state_solvers/test_adapt_vqe.py

```diff
@@ -68,6 +70,43 @@ def test_default(self):
         res = calc.solve(self.problem)
         self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
 
+    def test_param_shift(self):
+        """test for param_shift method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="param_shift")
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
+    def test_fin_diff(self):
+        """test for fin_diff method"""
+        solver = VQEUCCFactory(QuantumInstance(BasicAer.get_backend("statevector_simulator")))
+        grad = Gradient(grad_method="fin_diff", epsilon=1.0)
+        calc = AdaptVQE(self.qubit_converter, solver, gradient=grad)
+        res = calc.solve(self.problem)
+        self.assertAlmostEqual(res.electronic_energies[0], self.expected, places=6)
+
```

Comment:
I suggest that you also add unittests for the other gradient types which we started to test based on Julien's feedback. Even if they fail for now, once we debug that we should have tests to guard against future incompatibilities.